File: tests/rspec/pages/utilities/import_file_converter.rb
class ImportConverter < ControlPanelPage

  element :file_location, 'input[name=member_file]'
  element :delimiter, 'input[name=delimiter]'
  element :delimiter_special, 'input[name=delimiter_special]'
  element :enclosing_char, 'input[name=enclosure]'

  # Assign fields page
  element :field1, 'select[name=field_0]'
  element :field2, 'select[name=field_1]'
  element :field3, 'select[name=field_2]'
  element :field4, 'select[name=field_3]'

  # XML Code page
  element :xml_code, 'textarea.template-edit'

  def load
    self.open_dev_menu
    click_link 'Utilities'
    click_link 'File Converter'
  end
end

File: tests/cypress/cypress/elements/pages/utilities/Import_file_converter.js
import ControlPanel from '../ControlPanel'

class ImportConverter extends ControlPanel {
    constructor() {
        super()

        this.selectors = Object.assign(this.selectors, {



        	  "file_location" :'input[type!=hidden][name=member_file]',
			  "delimiter": 'div[class="checkbox-label__text"]',
			  "delimiter_special" :'input[type!=hidden][name=delimiter_special]',
			  "enclosing_char": 'input[type!=hidden][name=enclosure]',

              "send_it" : 'input[value="Convert File"]',



			  // Assign fields page
			  "field1" :'select[name=field_0]',
			  "field2" :'select[name=field_1]',
			  "field3" :'select[name=field_2]',
			  "field4" :'select[name=field_3]',
			  // XML Code page
			  "xml_code" :'textarea.template-edit',

              "send_it_2" : 'input[value="Assign Fields"]'

        })
    }

    load(){
    	 this.open_dev_menu()
    	 cy.get('body > .dropdown').find('a:contains("Utilities")').click()
    	 this.get('wrap').find('a:contains("File Converter")').click()

    }


    submit(fileName, fileType, selector){
        cy.get(selector).then(subject => {
                cy.fixture(fileName, 'base64')
                .then(Cypress.Blob.base64StringToBlob)
                .then(blob => {
                    const el = subject[0]
                    const testFile = new File([blob], fileName, { type: fileType })
                    const dataTransfer = new DataTransfer()
                    dataTransfer.items.add(testFile)
                    el.files = dataTransfer.files
                    console.log(el.files)
              })
        })
    }




}
export default ImportConverter;

